tempodb/encoding/vparquet3/block_iterator.go

```diff
@@ -75,6 +75,7 @@ func (i *rawIterator) Next(context.Context) (common.ID, parquet.Row, error) {
 	}
 
 	if errors.Is(err, io.EOF) {
+		i.pool.Put(rows[0])
```

Comment:
This fixes a "memory leak" where the buffer wasn't returned to the pool.  This actually has a significant effect on memory because this isn't happening once at the end, but on every loop of the multiblock iterator.  Once 1 block is exhausted it begins leaking a pooled buffer on every call. The better fix is to make the multiblock iterator stop calling Next() once an iter is exhausted, but it was going to be a lot more involved.